docs/RELEASE.md

```diff
@@ -0,0 +1,63 @@
+# Release
+
+This document describes how to release the Packet infrastructure provider.
+
+This is _not_ intended for regular users.
+
+This is normally performed by our CI system. However, there are important steps to take first.
+
+## How to Cut a Releas
+
+In order to cut a release, you must:
+
+1. If this is a new major or minor version - but **not** just a patch change - update [metadata.yaml](./metadata.yaml) to add it, and map it to the correct cluster-api contract version
+1. Modify `VERSION` so it includes only the version number to be released, in the format `vA.B.C`
```

Comment:
Is it still valid? At this point, I feel like we can use the current workflow (that has to be documented and probably update with the changes you on the Makefile targets) we have in `.github/workflows/release.yaml`